### DISCUSSION TITLE
store_read without throwing an error?

### DISCUSSION THREAD

**Discussion by simonbacquie** [datanymizer/datanymizer#126] [Q&A] [answered]
Hi,

I'm trying to use the key/value store feature to share row values across multiple tables, to preserve data integrity and all that. However, I have a situation where I'm trying to read data from the store to insert into one table, but the corresponding row in the other table may or may not exist. Therefore, the key is never written to the store. So, if I try to access that key using `store_read`, I get an error, and the entire data dump fails.

For example:

`user.first_name`
`user.email`
`attendee.first_name`
`attendee.email`
`attendee.user_id`

Let's say a User can be an Attendee - but Attendees can exist without being Users, and Users can exist without being Attendees. Ideally I would want to share values from User onto the Attendee table, but I can't guarantee that keys will exist in the store, because an Attendee can exist without having a `user_id`.

I hope that makes sense. Is there a solution? I looked around for ways to catch/suppress the error from within the Tera templating, but didn't have any luck.

Thanks!

**Comment by evgeniy-r** [comment]
Hello! We will discuss this.

**Comment by evgeniy-r** [comment, marked as answer]
We have added this feature to the `main` branch.
https://github.com/datanymizer/datanymizer/blob/main/docs/transformers.md#template

**Reply by simonbacquie** [reply]
Thanks so much for this - I see in the docs that there is now a default value option. I should be trying it out soon.
